cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/goals/CapacityGoal.java

```diff
@@ -435,6 +435,11 @@ private boolean isSwapAcceptableForCapacity(Replica sourceReplica, Replica desti
     double sourceReplicaUtilization = sourceReplica.load().expectedUtilizationFor(resource());
     double destinationReplicaUtilization = destinationReplica.load().expectedUtilizationFor(resource());
 
+    if (isUtilizationOverLimitAfterAddingLoad(sourceReplica.broker(), destinationReplicaUtilization)
+            || isUtilizationOverLimitAfterAddingLoad(destinationReplica.broker(), sourceReplicaUtilization)) {
+      return false;
+    }
+
     double sourceUtilizationDelta = destinationReplicaUtilization - sourceReplicaUtilization;
     return sourceUtilizationDelta > 0 ? isUtilizationUnderLimitAfterAddingLoad(sourceReplica.broker(), sourceUtilizationDelta)
```

Comment:
I think you could merge the logic here and do 
```
(isUtilizationUnderLimitAfterAddingLoad(sourceReplica.broker(), sourceUtilizationDelta) && !isUtilizationOverLimitAfterAddingLoad(sourceReplica.broker(), destinationReplicaUtilization)) : (isUtilizationUnderLimitAfterAddingLoad(destinationReplica.broker(), - sourceUtilizationDelta) && isUtilizationOverLimitAfterAddingLoad(destinationReplica.broker(), sourceReplicaUtilization))
```
and have a single return. Not sure if that logic is exactly correct, but that's my thought.

Reply:
I think we still have to check the intermediate utilization for both the source and destination brokers separately but I do like the idea of combining all of it into a single return - done!